### ISSUE TITLE
VST3 plugins and Components do not with "M1 MacOS and Elgato Wavelink" plugin

### ISSUE THREAD

**Issue by kuyazee** [werman/noise-suppression-for-voice#136]
I downloaded the latest plugins from [v1.03](https://github.com/werman/noise-suppression-for-voice/releases/tag/v1.03)  then had to trust the files using `sudo xattr -rd com.apple.quarantine $file_location_here` 

It finally shows up in Wavelink but the audio becomes muted when RNNoise plugin is added
![Screen Shot 2022-08-12 at 9 44 14 AM](https://user-images.githubusercontent.com/17846698/184269906-7a490fe9-6f40-4098-816d-e52a217b544d.png)

Here on this video I am speaking continously and it doesn't have anything when RNNoise is added and I'm confused as to why this is 

https://user-images.githubusercontent.com/17846698/184269965-b204c08a-194d-4e8b-865a-113046dddc36.mov

**Comment by werman** [comment]
Have you opened plugin settings? There is a bit of debug info there, also what if you reduce VAD threshold to zero?


> I'm thinking maybe can you add on your documentation how to build for MacOS so I can re-build using M1, cause that might be the issue

Should be same as "Compiling for x64:"

```
cmake -Bbuild -H. -GNinja -DCMAKE_BUILD_TYPE=Release
ninja -C build
```

Though the build made in CI should be compatible with both x64 and M1 CPUs.

**Comment by kuyazee** [comment]
Hello again @werman sorry for the late response

Yes I tried putting the VAD threshold to 0, even tried all the other settings to 0 or other mixes where others are in the middle or maxed out
 
![Screen Shot 2022-08-16 at 10 53 51 AM](https://user-images.githubusercontent.com/17846698/184788360-dbd49cbf-fb5e-4aeb-9401-7ead6c13d0c7.png)

I even closed & opened Wave Link and restarted my Macbook. But nothing seems to work. By any chance what app did you use this on and found that it worked? I'd like to validate it on my end maybe there's a misconfiguration somewhere

**Comment by blathering** [comment]
I have the same issue but I'm on an Intel Macbook (2018 Macbook Pro, Monterey)

To add another detail, when I add the plugin to WaveLink sound stops coming through and even if I remove the plugin sound doesn't come through until I restart WaveLink

I'm on WaveLink 1.5

**Comment by werman** [comment]
I don't have any macOS devices, so I cannot test things myself. It was tested in https://github.com/werman/noise-suppression-for-voice/issues/101#issuecomment-1166683352 with Ableton Live 10 and Wave Lab LE 11

**Comment by werman** [comment]
Also it seems it works with Audacity, see https://github.com/werman/noise-suppression-for-voice/pull/139

**Comment by blathering** [comment]
Is there anything I can do to help? Is there log output?

Contrary to the title of this issue my understanding is that the WaveLink doesn't support VST3 plugins on MacOS, only AU

**Comment by werman** [comment]
Does Elgato Wavelink work with just built-in mic, without their specific mic model?

**Comment by blathering** [comment]
Yes, you can add built-in Macbook mic as another channel. If I do that and then add the plugin the same thing happens, no audio. Audio continues to work on the other channels (without plugin) in that case.

**Comment by werman** [comment]
> Contrary to the title of this issue my understanding is that the WaveLink doesn't support VST3 plugins on MacOS, only AU

So which plugin did you copy and where?

**Comment by blathering** [comment]
rnnoise.component to /Library/Audio/Plug-Ins/Components/

**Comment by werman** [comment]
Also are you sure that you don't have 96kHz instead of 48kHz sampling rate?

**Comment by blathering** [comment]
There is no option to configure this on the Mac version of Wave Link so I'm unsure.

<img width="430" alt="image" src="https://user-images.githubusercontent.com/28418657/184985168-57f63a62-d042-41a3-ad24-bbfb789bca5b.png">
vs 
(what I assume is Windows from here: https://help.elgato.com/hc/en-us/articles/4404228579853-Elgato-Wave-XLR-Wave-Link-Settings-Overview)
![image](https://user-images.githubusercontent.com/28418657/184985211-cbc72cb5-a7f9-40c6-9d16-c59554a6b203.png)

**Comment by werman** [comment]
Maybe it's in Mac's mic settings?

> Yes, you can add built-in Macbook mic as another channel. If I do that and then add the plugin the same thing happens, no audio. Audio continues to work on the other channels (without plugin) in that case.

Unfortunately the app itself refuses to launch without their branded device connected.

**Comment by blathering** [comment]
I don't have the same configuration settings for the mac mic, only available for the mic connected via my Wave XLR.

**Comment by werman** [comment]
I mean the settings available in the system itself, I think the macOS itself has some sound settings.

**Comment by blathering** [comment]
I can't see any settings relating to the sampling rate at system level

**Comment by werman** [comment]
Ugh, the only other thing to check without me adding additional logs, is to check with another plugin host which supports AU plugins.

I think I'll add an option to log into file, though I'll do it a bit later.

**Comment by blathering** [comment]
Just tested and the Audio Unit seems to work in GarageBand

**Comment by kuyazee** [comment]
Can confirm on my end that it also works for Garageband.

Is there anything I could help with maybe I can send some logs or something.

This could be a limitation for Wave Link as well. I'll try to raise it on Reddit and their Customer Support

**Comment by werman** [comment]
@kuyazee So are you using VST3 or AU? Since @blathering tells that Wave Link supports only AU.

**Comment by fgauna12** [comment]
I'm trying to use AU and having similar issues. Debug statistics don't show anything.

**Comment by kuyazee** [comment]
@werman I tried to use both

**Comment by TomMarg** [comment]
+1

**Comment by tpetersons** [comment]
Same problem, cannot get RNNoise working with macbook pro with M1 processor.
My elgato microphone is unusable without noise suppression, im surprised that this is the only plugin out there and is not working. Any help?

**Comment by bselwe** [comment]
@tpetersons Did you figure out any solution for noise suppression on Mac M1 maybe?

**Comment by bselwe** [comment]
@werman I could try to debug/produce logs or we could have a session to understand this if that would be any helpful. I am using MacOS Ventura 13.0 with Elgato Wave 3 mic. Do you happen to know if there is a way to report the failing logs?

> Also are you sure that you don't have 96kHz instead of 48kHz sampling rate?

I can confirm that I am using 48kHz sampling rate for the Wave 3. It defaults to 96kHz in the system settings though. This may be changed in Audio MIDI Setup app: 
![CleanShot 2022-12-08 at 01 03 58@2x](https://user-images.githubusercontent.com/26686598/206323707-b676d459-fc7a-4725-8f98-17a74a13f71d.png)

**Comment by tmikaeld** [comment]
I found this in the Console logs:
Sandbox: logd_helper(159) deny(1) file-read-data ~/Library/Audio/Plug-Ins/Components/rnnoise.component/Contents/MacOS/rnnoise

I tried allowing it in the quarantine, but that didn't help, might be the reason it doesn't work.

**Comment by tmikaeld** [comment]
I've tried with premium AI noise suppression plugin https://www.waves.com/plugins/clarity-vx and it does not work either, same issue - input stops working as soon as plugin is active.

**Comment by vilinskyy** [comment]
I have the same problem here. Adding rnnoise in any way and enabling it just kills any input from the mic, until I delete rnnoise and relaunch wavelink.

Would love to help to fix it.

<img width="392" alt="Screenshot 2022-12-20 at 18 01 37" src="https://user-images.githubusercontent.com/11735463/208735005-7aba0ea2-b3fe-4688-843e-274dda0a5c5c.png">

**Comment by hengyu95** [comment]
> I have the same problem here. Adding rnnoise in any way and enabling it just kills any input from the mic, until I delete rnnoise and relaunch wavelink.

Exact problem here

**Comment by littke** [comment]
same problem

**Comment by littke** [comment]
Got it to work in Elgato Wave Link by combining everything in this thread:

0. Download from GitHub
1. Set mic to 48khz in Audio MIDI Setup (native application)
2. `mv rnnoise.component /Library/Audio/Plug-Ins/Components/`
3. `/Library/Audio/Plug-Ins/Components/`
`sudo xattr -rd com.apple.quarantine rnnoise.component`
4. Add it in Wave Link

**Comment by blathering** [comment]
Can confirm this is working for me too.

2018 Intel Macbook Pro, macOS Ventura 13.4.1, Wave Link v1.8
